### PR TITLE
Initialize container in WebService environment

### LINKED ISSUE #14842: Dosen't send email on state change using WS since 1.7.6
After updating to 1.7.6...
When changing ordre "current_state" using webservice, emails is not sent tho ordre state changes.

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x 
| Description?  | Initialize container in WebService environment
| Type?         | bug fix
| Category?     | WS
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | Fixes #14842 
| How to test?  | * Try the script given in issue #14842#issuecomment-514593911

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/14896)

### CODE REVIEW

File: classes/controller/WSController.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    /**
+     * @return bool|void
+     *
+     * @throws Exception
+     */
+    public function checkAccess()
+    {
+        throw new Exception('Method checkAccess() not implemented in WSController');
```
Comment (PierreRambaud) [inline classes/controller/WSController.php:51]: So an exception is always raised when the controller is running :thinking: 
https://github.com/PrestaShop/PrestaShop/blob/develop/classes/controller/Controller.php#L273

Comment (PierreRambaud) [reply classes/controller/WSController.php:51]: We maybe need to create  a default value sometimes, depending on webservice context.

Comment (Progi1984) [reply classes/controller/WSController.php:51]: No, because in our case, we construct and init the controller but there is no run.

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @Progi1984,

When I try to navigate to this link: `http://shop.com/api/`
I have this error 
```Fatal error: Uncaught Symfony\Component\Debug\Exception\FatalThrowableError: Cannot use object of type Symfony\Component\HttpFoundation\ParameterBag as array in C:\xampp\htdocs\develop\src\Adapter\Security\Admin.php:91 
Stack trace: 
#0 C:\xampp\htdocs\develop\modules\gamification\vendor\symfony\event-dispatcher\Debug\WrappedListener.php(115): PrestaShop\PrestaShop\Adapter\Security\Admin->onKernelRequest(Object(Symfony\Component\HttpKernel\Event\GetResponseEvent), 'kernel.request', Object(Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher)) 
#1 C:\xampp\htdocs\develop\modules\gamification\vendor\symfony\event-dispatcher\EventDispatcher.php(212): Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(Object(Symfony\Component\HttpKernel\Event\GetResponseEvent), 'kernel.request', Object(Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher)) 
#2 C:\xampp\htdocs\develop\modules\gamification\vendor\symfony\event-dispatcher\EventDispatcher.php(44): Symfony\Component\EventDispatcher\EventDispatcher->doDispa in C:\xampp\htdocs\develop\src\Adapter\Security\Admin.php on line 91
```

In the branch 1.7.6.x => This XML file is well displayed.

Thanks!

**Comment by Progi1984** [comment]
@khouloudbelguith I test a new way for correcting the bug. Thanks for the feedback ;)

**Comment by Progi1984** [comment]
Replaced by #14899

**Pull request closed without merging**